### ISSUE TITLE
`no-length-as-slice-end`: Forbid use `.length` as the `end` argument of `{Array,String,TypedArray}#slice()`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2387]
### Description

Forbid use `.length` as the `end` argument of `{Array,String,TypedArray}#slice()`

### Fail

```js
array.slice(start, array.length)
```

```js
array?.slice(start, array.length)
```

### Pass

```js
array.slice(start)
```

```js
array?.slice(start)
```

**Comment by fisker** [comment]
accepted
